packages/web/src/components/track/PremiumTrackSection.tsx

```diff
@@ -0,0 +1,371 @@
+
+const messages = {
+  howToUnlock: 'HOW TO UNLOCK',
+  unlockCollectibleGatedTrack:
+    'To unlock this track, you must link a wallet containing a collectible from:',
+  unlockedCollectibleGatedTrackPrefix: 'A Collectible from ',
+  unlockedCollectibleGatedTrackSuffix:
+    ' was found in a linked wallet. This track is now available.',
+  goToCollection: 'Go To Collection',
+  sendTip: 'Send Tip',
+  followArtist: 'Follow Artist',
+  unlockFollowGatedTrackPrefix: 'Follow ',
+  unlockedFollowGatedTrackPrefix: 'Thank you for following ',
+  unlockTipGatedTrackPrefix: 'Send ',
+  unlockTipGatedTrackSuffix: ' a tip',
+  unlockedTipGatedTrackPrefix: 'Thank you for tipping ',
+  unlockedSpecialAccessGatedTrackSuffix: '! This track is now available.',
+  unlocked: 'UNLOCKED',
+  unlocking: 'Unlocking'
+}
+
+type PremiumTrackAccessSectionProps = {
+  premiumConditions: PremiumConditions
+  followee: Nullable<User>
+  tippedUser: Nullable<User>
+}
+
+const LockedPremiumTrackSection = ({
+  premiumConditions,
+  followee,
+  tippedUser
+}: PremiumTrackAccessSectionProps) => {
+  const dispatch = useDispatch()
+  const sendStatus = useSelector(getSendStatus)
+  const previousSendStatus = usePrevious(sendStatus)
+  const [isUnlocking, setIsUnlocking] = useState(false)
+
+  // Set unlocking state if send tip is successful and user closed the tip modal.
+  useEffect(() => {
+    if (previousSendStatus === 'SUCCESS' && sendStatus === null) {
+      setIsUnlocking(true)
+      // Poll discovery to get user's premium content signature for this track.
+      // dispatch(refreshPremiumTrack())
+    }
+  }, [previousSendStatus, sendStatus])
+
+  const handleSendTip = useCallback(() => {
+    dispatch(beginTip({ user: tippedUser, source: 'trackPage' }))
+  }, [dispatch, tippedUser])
+
+  const handleFollow = useCallback(() => {
+    if (premiumConditions.follow_user_id) {
+      dispatch(
+        socialActions.followUser(
+          premiumConditions.follow_user_id,
+          FollowSource.TRACK_PAGE
+        )
+      )
+      // Set unlocking state if user has clicked on button to follow artist.
+      setIsUnlocking(true)
+      // Poll discovery to get user's premium content signature for this track.
+      // dispatch(refreshPremiumTrack())
+    }
+  }, [dispatch, premiumConditions])
+
+  const handleGoToCollection = useCallback(() => {
+    const { chain, address, externalLink } =
+      premiumConditions.nft_collection ?? {}
+    if (chain === Chain.Eth && 'slug' in premiumConditions.nft_collection!) {
+      const url = `https://opensea.io/collection/${premiumConditions.nft_collection.slug}`
+      window.open(url, '_blank')
+    } else if (chain === Chain.Sol) {
+      const explorerUrl = `https://explorer.solana.com/address/${address}`
+      const url = externalLink ? new URL(externalLink).hostname : explorerUrl
+      window.open(url, '_blank')
+    }
+  }, [premiumConditions])
+
+  const renderLockedDescription = useCallback(() => {
```

Comment:
thinking about separating this by gate type rather than by UI component...
maybe in another PR